app/controllers/lti_assignment_users_controller.rb

```diff
@@ -0,0 +1,221 @@
+include Lti2Commons
+include Signer
+include MessageSupport
+include OAuth::OAuthProxy
+class LtiAssignmentUsersController < ApplicationController
+  protect_from_forgery :except => :create
+
+  def action_allowed?
+    case params[:action]
+      when 'create', 'back_to_lms'
+        return true
+    end
+  end
+
+  def create
+    (@error_code, @message) = pre_process_tenant
+    if @error_code == 200 # 200 means pre processing was successful
+      (user, @error_code, @message) = lti_login()
+      if(user)
+        assignment_id = params['custom_assignment_id'] # assignment_id needs to be passed as a
+        assignment = Assignment.find(assignment_id)
+        if assignment
+          # Assignment number passed in the request is valid.
+          assignment_participant = AssignmentParticipant.find_by_user_id_and_assignment_id(user.id,assignment_id)
+          lis_result_sourcedid = params["lis_result_sourcedid"]; # sourcedid is used to send grades back
+          @tenant.lis_outcome_service_url = params['lis_outcome_service_url'];
+          unless assignment_participant
+            # Assignment_Participant not found in participant table and assignment exists. So add user to participant table.
```

Comment:
It seems to me that checking for and/or adding assignment_participant should be done in a different method.